metals/src/main/scala/scala/meta/internal/metals/BuildTargets.scala

```diff
@@ -239,9 +250,23 @@ final class BuildTargets() {
    */
   def inferBuildTarget(
       source: AbsolutePath
-  ): Option[BuildTargetIdentifier] =
-    if (!source.isDependencySource(workspace)) None
-    else Try(unsafeInferBuildTarget(source)).getOrElse(None)
+  ): Option[BuildTargetIdentifier] = {
+    if (source.isDependencySource(workspace)) {
+      Try(unsafeInferBuildTarget(source)).getOrElse(None)
```

Comment:
`.toOption`?

Reply:
This code is unchanged from before, it just moved. I think `toOption.flatten` is a valid refactoring but it's unrelated to this PR.